agent/consul/config_endpoint.go

```diff
@@ -542,127 +443,266 @@ func (c *ConfigEntry) ResolveServiceConfig(args *structs.ServiceConfigRequest, r
 				}
 			}
 
-			// First store all upstreams that were provided in the request
-			for _, sid := range upstreamIDs {
-				if _, ok := seenUpstreams[sid]; !ok {
-					seenUpstreams[sid] = struct{}{}
+			// Fetch all relevant config entries. We will generate a
```

Comment:
The body of this `blockingQuery` call was changed into two steps:
1. Get all config entries we will need.
2. Smoosh those config entries together to fabricate the result.

Iterations after the first one will stop after (1) and check if any of the config entries collected differ from the prior iteration. If they do it returns the new data, otherwise it signals to the RPC side that it is a `not-changed` scenario.